src/SAWScript/Crucible/LLVM/X86.hs

```diff
@@ -700,29 +702,35 @@ assertPointsTo env tyenv nameEnv (LLVMPointsTo _ cond tptr texpected) = do
 
 -- | Gather and run the solver on goals from the simulator.
 checkGoals ::
-  X86Sim ()
-checkGoals = do
-  sym <- use x86Sym
-  opts <- use x86Options
-  sc <- use x86SharedContext
+  Sym ->
+  Options ->
+  SharedContext ->
+  ProofScript SatResult ->
+  TopLevel ()
+checkGoals sym opts sc tactic = do
   gs <- liftIO $ getGoals sym
   liftIO . printOutLn opts Info $ mconcat
     [ "Simulation finished, running solver on "
     , show $ length gs
     , " goals"
     ]
-  liftIO . forM_ gs $ \g ->
-    do
-      term <- gGoal sc g
-      (mb, stats) <- proveUnintSBV z3 [] Nothing sc term
-      printOutLn opts Info $ ppStats stats
-      case mb of
-        Nothing -> printOutLn opts Info "Goal succeeded"
-        Just ex -> do
-          fail $ mconcat
-            ["Failure (", show $ gLoc g
-            , "): ", show $ gMessage g
-            , "\nCounterexample: " <> show ex
-            ]
-    `catch` \(X86Error e) -> fail $ "Failure, error: " <> e
+  forM_ (zip [0..] gs) $ \(n, g) -> do
```

Comment:
is it possible to extract this and use it both here and in `verifyObligations` in `SAWScript.Crucible.LLVM.Builtins`?